### DISCUSSION TITLE
Accessing client.meshtastic.org

### DISCUSSION THREAD

**Discussion by spectrumcomputing** [meshtastic/Discussions#114] [ESP32]
Hello,

I'm using a 'LILYGO® TTGO LoRa32 V2.1_1.6', and I'm using 2.5.12 on Android. When I flashed the device I chose not to install the web client. I have flashed it a number of times, and have previously installed the web client, but I always tick the erase box before flashing.

I've configured via the Android app to use WiFi, and have switched off BlueTooth.  I'm connecting on 192.168.1.147

If I connect to that IP address in my Chrome browser I get:

```
Web server is running.

The content you are looking for can't be found. Please see: [FAQ](https://meshtastic.org/docs/software/web-client/).

[admin](http://192.168.1.147/admin)
```
When I visit client.meshtastic.org I have tried leaving the address as meshtastic.local, and tried 192.168.1.147

Both result in just a 'Messages: Loading...', and nothing every happens.

What am I doing wrong please?

Thanks 

Peter M0LMG

This is the device report:

```
{
  "data": {
    "airtime": {
      "channel_utilization": 0,
      "periods_to_log": 8,
      "rx_all_log": [0, 0, 0, 0, 0, 0, 0, 0],
      "rx_log": [0, 0, 0, 0, 0, 0, 0, 0],
      "seconds_per_period": 3600,
      "seconds_since_boot": 700,
      "tx_log": [2578, 0, 0, 0, 0, 0, 0, 0],
      "utilization_tx": 0.0716111063957214
    },
    "device": {
      "reboot_counter": 31
    },
    "memory": {
      "fs_free": 1028096,
      "fs_total": 1048576,
      "fs_used": 20480,
      "heap_free": 32072,
      "heap_total": 184952,
      "psram_free": 0,
      "psram_total": 0
    },
    "power": {
      "battery_percent": 100,
      "battery_voltage_mv": 4192,
      "has_battery": "true",
      "has_usb": "false",
      "is_charging": "false"
    },
    "radio": {
      "frequency": 869.525024414062,
      "lora_channel": 1
    },
    "wifi": {
      "ip": "192.168.1.147",
      "rssi": -44
    }
  },
  "status": "ok"
}

```

**Comment by garthvh** [comment]
If you are connecting to the device ip that would be accessing the on device web client which you chose not to install

**Comment by ianmcorvidae** [comment]
To access it via client.meshtastic.org you probably need to get it working with SSL. Go to https://192.168.1.147 and accept the self signed certificate in your browser of choice. After you've done that you should be able to use it with the hosted client.

**Comment by spectrumcomputing** [comment]
@garthvh,

Thanks for the reply.  I'm sorry, but I don't understand the answer.  Are you saying that to add a client to client.meshtastic.org,  I have to install the web client installed on my device too?

Please can I make a request, that tickets are not closed until the reporter has had a change to read, review and ask for any clarification? (For example 48 hours without a reply)?

Thanks so much for the project, I'm having so much fun!

Peter M0LMG

**Comment by spectrumcomputing** [comment]
Thanks for the reply @ianmcorvidae,

I've tried on 'Chrome' based browsers, and on Edge I just get this: I see no option to accept a self-signed certificate.

![image](https://github.com/user-attachments/assets/d220b4ea-8c20-45cf-b1af-535430dfc04f)

![image](https://github.com/user-attachments/assets/c3ef23c9-2f4e-474d-912f-eb6af41ca429)

I can access over http. Thanks

**Reply by ianmcorvidae** [reply]
Hm, odd. The node should be able to generate itself a self-signed certificate, and that would be required in this case -- client.meshtastic.org is only accessible over HTTPS, so the connection to the node must also be HTTPS due to browser restrictions. If you have the option, it could be worth watching the serial log for the node, first while it's booting and then when you attempt to connect to it via HTTP. For the first part, what you should be seeing is some subset of the following:

> Checking if we have a saved SSL Certificate
> Existing SSL Certificate found!
> Creating the certificate. This make take a while. Please wait
> Creating the certificate failed
> Creating the certificate was successful

When connecting, I'm not sure what would be logged exactly. A connection reset is an odd error -- normally you'd get a page telling you the certificate wasn't secure or similar instead, but it seems to not be getting to that.

**Reply by spectrumcomputing** [reply]
Thanks @ianmcorvidae,

I will install Python so I can try the serial connection. I will post anything that looks like it may help.

I'll also try a reinstall.

Could I just check that I don't need to choose to install the web client on the device when going through the install process to be able to still use client.meshtastic.org?

Peter

**Reply by ianmcorvidae** [reply]
Yes, that's correct -- you shouldn't need to have the web client installed directly on the device in order to use the hosted web client at client.meshtastic.org. Garth's comment was only an explanation of why it says "this content could not be found" when connecting directly.

As far as viewing the serial log, if you're using the web flasher and web serial, there should also be a serial log viewer there. Not that using the CLI isn't also an option, but if you run into issues there, also worth a try.

**Reply by spectrumcomputing** [reply]
Thanks, I will take a look in the morning.

**Comment by spectrumcomputing** [comment]
Hi @ianmcorvidae,

Nothing unusual in the log, so will try a reflash.

Peter M0LMG

```
DEBUG | ??:??:?? 140 [Power] Battery: usbPower=1, isCharging=1, batMv=4237, batPct=100

DEBUG | ??:??:?? 160 [Power] Battery: usbPower=0, isCharging=0, batMv=4194, batPct=100

INFO | ??:??:?? 165 [DeviceTelemetry] Send: air_util_tx=0.037611, channel_utilization=0.000000, battery_level=100, voltage=4.194000, uptime=165

DEBUG | ??:??:?? 165 [DeviceTelemetry] Partially randomized packet id 582811637

DEBUG | ??:??:?? 165 [DeviceTelemetry] updateTelemetry LOCAL

DEBUG | ??:??:?? 165 [DeviceTelemetry] Node status update: 1 online, 1 total

INFO | ??:??:?? 165 [DeviceTelemetry] Send packet to phone

INFO | ??:??:?? 165 Tell client we have new packets 7

DEBUG | ??:??:?? 166 [SerialConsole] FromRadio=STATE_SEND_PACKETS

DEBUG | ??:??:?? 166 [SerialConsole] phone downloaded packet (id=0x22bcfff5 fr=0x0c to=0xff, WantAck=0, HopLim=5 Ch=0x0 Portnum=67 priority=10)

��649�����"C�d?5�@%=(�5���"HX

DEBUG | ??:??:?? 180 [Power] Battery: usbPower=1, isCharging=1, batMv=4229, batPct=100

DEBUG | ??:??:?? 200 [Power] Battery: usbPower=1, isCharging=1, batMv=4238, batPct=100

DEBUG | ??:??:?? 220 [Power] Battery: usbPower=0, isCharging=0, batMv=4195, batPct=100

INFO | ??:??:?? 225 [DeviceTelemetry] Send: air_util_tx=0.037611, channel_utilization=0.000000, battery_level=100, voltage=4.195000, uptime=225

DEBUG | ??:??:?? 225 [DeviceTelemetry] Partially randomized packet id 198534134

DEBUG | ??:??:?? 225 [DeviceTelemetry] updateTelemetry LOCAL

DEBUG | ??:??:?? 225 [DeviceTelemetry] Node status update: 1 online, 1 total

INFO | ??:??:?? 225 [DeviceTelemetry] Send packet to phone

INFO | ??:??:?? 225 Tell client we have new packets 8

DEBUG | ??:??:?? 226 [SerialConsole] FromRadio=STATE_SEND_PACKETS

DEBUG | ??:??:?? 226 [SerialConsole] phone downloaded packet (id=0x0bd563f6 fr=0x0c to=0xff, WantAck=0, HopLim=5 Ch=0x0 Portnum=67 priority=10)

��649�����"C�dq=�@%=(�5�c�HX

DEBUG | ??:??:?? 240 [Power] Battery: usbPower=1, isCharging=1, batMv=4246, batPct=100

DEBUG | ??:??:?? 260 [Power] Battery: usbPower=1, isCharging=1, batMv=4230, batPct=100

DEBUG | ??:??:?? 280 [Power] Battery: usbPower=1, isCharging=1, batMv=4239, batPct=100

INFO | ??:??:?? 285 [DeviceTelemetry] Send: air_util_tx=0.037611, channel_utilization=0.000000, battery_level=101, voltage=4.239000, uptime=285

DEBUG | ??:??:?? 285 [DeviceTelemetry] Partially randomized packet id 1314292727

DEBUG | ??:??:?? 285 [DeviceTelemetry] updateTelemetry LOCAL

DEBUG | ??:??:?? 285 [DeviceTelemetry] Node status update: 1 online, 1 total

INFO | ??:??:?? 285 [DeviceTelemetry] Send packet to phone

INFO | ??:??:?? 285 Tell client we have new packets 9

DEBUG | ??:??:?? 286 [SerialConsole] FromRadio=STATE_SEND_PACKETS

DEBUG | ??:??:?? 286 [SerialConsole] phone downloaded packet (id=0x4e5683f7 fr=0x0c to=0xff, WantAck=0, HopLim=5 Ch=0x0 Portnum=67 priority=10)

��649�����"Ce㥇@%=(�5��VNHX

DEBUG | ??:??:?? 300 [Power] Battery: usbPower=1, isCharging=1, batMv=4259, batPct=100
```

**Comment by spectrumcomputing** [comment]
@ianmcorvidae,

Hi Ian.  I tried reflashing and still no job, I just don't get the unsecure certificate when I try and access the devices IP address.  I've resolved the issue by connecting via BlueTooth, or Serial with the USB cable.

Thanks for trying to help.
